### DISCUSSION TITLE
NER scorer is always zero

### DISCUSSION THREAD

**Discussion by ghost** [explosion/spaCy#8195] [Help: Coding & Implementations] [answered]
Hi, I am trying to use a Spacy NER custom model, but then scorer prints all in zero. I verified my training data with:
!python -m spacy debug data config.cfg --paths.train ./NERtrain.spacy --paths.dev ./NERval.spacy
And get evertything kind of ok. So I have no idea why is not showing any increase in metrics. Thanks in advance for the help.

```
============================ Data file validation ============================
✔ Corpus is loadable
✔ Pipeline can be initialized with data

=============================== Training stats ===============================
Language: es
Training pipeline: tok2vec, ner
180 training docs
20 evaluation docs
⚠ 17 training examples also in evaluation data
⚠ Low number of examples to train a new pipeline (180)

============================== Vocab & Vectors ==============================
ℹ 52356 total word(s) in the data (5296 unique)
ℹ 500000 vectors (500000 unique keys, 300 dimensions)
⚠ 7805 words in training data without vectors (0.15%)

========================== Named Entity Recognition ==========================
ℹ 0 new label(s), 0 existing label(s)
0 missing value(s) (tokens with '-' label)
✔ Good amount of examples for all labels
✔ Examples without occurrences available for all labels
✔ No entities consisting of or starting/ending with whitespace
✔ No entities consisting of or starting/ending with punctuation

================================== Summary ==================================
✔ 6 checks passed
⚠ 3 warnings
This is my current scorer:

=========================== Initializing pipeline ===========================
[2021-05-25 20:40:42,005] [INFO] Set up nlp object from config
[2021-05-25 20:40:42,013] [INFO] Pipeline: ['tok2vec', 'ner']
[2021-05-25 20:40:42,015] [INFO] Created vocabulary
[2021-05-25 20:40:43,148] [INFO] Added vectors: es_core_news_lg
[2021-05-25 20:40:43,148] [INFO] Finished initializing nlp object
[2021-05-25 20:40:45,931] [INFO] Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['tok2vec', 'ner']
ℹ Initial learn rate: 0.001
E    #       LOSS TOK2VEC  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE 
---  ------  ------------  --------  ------  ------  ------  ------
  0       0          0.00    224.00    0.00    0.00    0.00    0.00
  1     200          0.06    245.57    0.00    0.00    0.00    0.00
  2     400          0.00      0.00    0.00    0.00    0.00    0.00
```

**Comment by polm** [comment]
Well, this looks like a bug in the debug data command that it didn't warn you, but if you look at this line it seems like you have no NER labels.

> ℹ 0 new label(s), 0 existing label(s)

Are you sure your annotations are included in your training data?

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/49178323/119605521-8f408580-bdb6-11eb-8fd0-f3e7e0d3e62d.png)
Just a small portion of the data's annotations has a None span. But the rest is included...

**Reply by polm** [reply]
Just to check, what version of spaCy are you using?

**Reply by ghost** [reply]
'3.0.5'

**Reply by ghost** [reply]
Could be something related to the config file? I did see another post that changing max_lenght to a higher number with a large document fixes the problem?
This is my current config.cfg file
[paths]
train = null
dev = null
vectors = null
init_tok2vec = null

[system]
gpu_allocator = null
seed = 0

[nlp]
lang = "es"
pipeline = ["tok2vec","ner"]
batch_size = 1000
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}

[components]

[components.ner]
factory = "ner"
moves = null
update_with_oracle_cut_size = 100

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = true
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = ${components.tok2vec.model.encode.width}
upstream = "*"

[components.tok2vec]
factory = "tok2vec"

[components.tok2vec.model]
@architectures = "spacy.Tok2Vec.v2"

[components.tok2vec.model.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = ${components.tok2vec.model.encode.width}
attrs = ["ORTH","SHAPE"]
rows = [5000,2500]
include_static_vectors = true

[components.tok2vec.model.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = 256
depth = 8
window_size = 1
maxout_pieces = 3

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 20000000
gold_preproc = false
limit = 0
augmenter = null

[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
accumulate_gradient = 1
patience = 1600
max_epochs = 0
max_steps = 20000
eval_frequency = 200
frozen_components = []
before_to_disk = null

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2
get_length = null

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001
t = 0.0

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001
learn_rate = 0.001

[training.score_weights]
ents_per_type = null
ents_f = 1.0
ents_p = 0.0
ents_r = 0.0

[pretraining]

[initialize]
vectors = "es_core_news_lg"
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null
before_init = null
after_init = null

[initialize.components]

[initialize.tokenizer]

**Reply by ghost** [reply]
The problems was overlapping entiites. Once I used a model for each wanted entity problem is solved!3

**Reply by polm** [reply]
Glad you figured it out, thanks for the update!

**Comment by polm** [comment, marked as answer]
For future reference, the issue was overlapping entities, and it was solved by training separate NER models.
